closure-typing/calculator.js

```diff
@@ -1,5 +1,21 @@
-function calculator() {
-  // your code here
+function calculator(str) {
+  var mathFuncList = {
+    '+' : (a, b) => {return +a + +b},
+    '-' : (a, b) => {return a - b},
+    '*' : (a, b) => {return a * b},
+    '/' : (a, b) => {return a / b}
+  }
+  arr = str.replace(/([+\-*/])/g, ' $1 ').split(' ')
+
+  while (arr.length > 1){
+    i = arr.indexOf('*')
+    i < 0 ? i = arr.indexOf('/') : i
```

Comment:
Not sure what is happening there, but smth telling me that it can be simplified, but if not nvm